src/components/common/DynamicTabs/useTabs.ts

```diff
@@ -0,0 +1,197 @@
+        let _tabs: DynamicTabType[]
+        try {
+            parsedTabsData = JSON.parse(persistedTabsData)
+            _tabs = persistedTabsData ? parsedTabsData.data : tabs
+        } catch (err) {
+            _tabs = tabs
+        }
+
+        if (_tabs.length > 0) {
+            const tabNames = _tabs.map((_tab) => _tab.name)
+            initTabsData.forEach((_initTab, idx) => {
+                if (!tabNames.includes(_initTab.name)) {
+                    _tabs.push(populateInitTab(_initTab, idx))
+                }
+            })
+        } else {
+            initTabsData.forEach((_initTab, idx) => {
+                _tabs.push(populateInitTab(_initTab, idx))
+            })
+        }
+
+        localStorage.setItem('persisted-tabs-data', stringifyData(_tabs, parsedTabsData))
+        setTabs(_tabs)
+    }
+
+    const addTab = (
+        idPrefix: string,
+        kind: string,
+        name: string,
+        url: string,
+        positionFixed?: boolean,
+        iconPath?: string,
+    ): boolean => {
+        if (!name || !url || !kind) return
+
+        const title = `${kind}/${name}`
+        let alreadyAdded = false
+        const _id = `${idPrefix}-${title}`
+        const _tabs = tabs.map((tab) => {
+            tab.isSelected = false
+            if (tab.title.toLowerCase() === title.toLowerCase() && tab.id === _id) {
+                tab.isSelected = true
+                tab.url = url
+                alreadyAdded = true
+            }
+            return tab
+        })
+
+        if (!alreadyAdded) {
+            _tabs.push(populateTabData(_id, title, url, true, title, positionFixed, iconPath))
+        }
+
+        localStorage.setItem('persisted-tabs-data', stringifyData(_tabs))
+        setTabs(_tabs)
+        return true
+    }
+
+    const removeTabByIdentifier = (title: string): string => {
+        let pushURL = ''
+        let selectedRemoved = false
+        const _tabs = tabs.filter((tab) => {
+            if (tab.title.toLowerCase() == title.toLowerCase()) {
+                selectedRemoved = tab.isSelected
+                return false
+            }
+            return true
+        })
+
+        if (selectedRemoved) {
+            _tabs[0].isSelected = true
+            pushURL = _tabs[0].url
+        }
+
+        localStorage.setItem('persisted-tabs-data', stringifyData(_tabs))
+        setTabs(_tabs)
+        return pushURL
+    }
+
+    const markTabActiveByIdentifier = (idPrefix: string, name: string, kind?: string, url?: string) => {
+        if (!name) return
+
+        let isTabFound = false
+        let title = name
+        if (kind) {
+            title = kind + '/' + name
+        }
+
+        const _id = `${idPrefix}-${title}`
+        const _tabs = tabs.map((tab) => {
+            tab.isSelected = false
+            if (tab.title.toLowerCase() === title.toLowerCase() && tab.id === _id) {
+                tab.isSelected = true
+                tab.url = url || tab.url
```

Comment:
can we leverage `??` here?

Reply:
Actually, url can be empty so used `||` which checks for empty string as well.